src/tdydiscretization.c

```diff
@@ -0,0 +1,159 @@
+#include <private/tdycoreimpl.h>
+
+/* -------------------------------------------------------------------------- */
+/// Creates a PETSc global vector
```

Comment:
Here's an opportunity to include a more precise description of what we mean when we talk about "global", "local", and "natural" vectors. In other words, what does the `i`th degree of freedom in this vector mean? Once we have these documented, it becomes clearer which ones are appropriate to use where.

Reply:
Local: A local vector is serial, and lays out data according to the local section. This section assigns dofs to the points of the local Plex.

Global: A global vector is parallel, and lays out data according to the global section. This section assigns dofs to the points of the local Plex, but leaves out any point that is in the pointSF because this means it is not owned by the process.

Natural: A natural vector is a permutation of a global vector, usually to match the input ordering.

Reply:
Right, these are familiar to me. But documenting them with markup can give people some confidence. And because we have a specific family of discretizations, we can be even more specific bout layouts if we want.

Reply:
Ah, good point. Do you want this documentation in PETSc or TDyCore?

Reply:
I was thinking TDyCore, because we're using these things in specific ways. These functions are already documented in PETSc, right?

Reply:
Somewhat yes. Okay, where should I put this documentation do you think?

Reply:
I will copy/paste @knepley's definitions in my PR and make a new commit.

Reply:
I was just telling @bishtgautam that I think he should mention some of this stuff in our Doxygen comments for the wrappers he's adding. I don't think you need to do any more documentation unless you think adding more detail to `DMCreate{Global,Local,Natural}Vector` in PETSc would be constructive (which I suppose it could be!). Thanks, though!